src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,326 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Remoting.Channels;
+using System.Runtime.Remoting.Channels.Ipc;
+using System.Threading.Tasks;
+using System.Windows.Automation;
+using EnvDTE;
+using Microsoft.Win32;
+
+using DteProject = EnvDTE.Project;
+using Process = System.Diagnostics.Process;
+
+namespace Roslyn.VisualStudio.Test.Utilities
+{
+    public class IntegrationHost : IDisposable
+    {
+        internal static readonly string VsProductVersion = Settings.Default.VsProductVersion;
+        internal static readonly string VsProgId = $"VisualStudio.DTE.{VsProductVersion}";
+
+        internal static readonly string Wow6432Registry = Environment.Is64BitProcess ? "WOW6432Node" : string.Empty;
+        internal static readonly string VsRegistryRoot = Path.Combine("SOFTWARE", Wow6432Registry, "Microsoft", "VisualStudio", VsProductVersion);
+        internal static readonly string VsConfigRootSubKey = $"{VsRegistryRoot}_Config";
+        internal static readonly string VsInitConfigSubKey = Path.Combine(VsConfigRootSubKey, "Initialization");
+
+        internal static readonly string VsCommon7Folder = Path.GetFullPath(IntegrationHelper.GetRegistryKeyValue(Registry.LocalMachine, VsRegistryRoot, "InstallDir").ToString());
+        internal static readonly string VsUserFilesFolder = Path.GetFullPath(IntegrationHelper.GetRegistryKeyValue(Registry.CurrentUser, VsInitConfigSubKey, "UserFilesFolder").ToString());
+
+        internal static readonly string VsExeFile = Path.Combine(VsCommon7Folder, "devenv.exe");
+        internal static readonly string VsLaunchArgs = $"{(string.IsNullOrWhiteSpace(Settings.Default.VsRootSuffix) ? "/log" : $"/rootsuffix {Settings.Default.VsRootSuffix}")} /log";
+
+        internal static readonly string VsStartServiceCommand = "Tools.StartIntegrationTestService";
+        internal static readonly string VsStopServiceCommand = "Tools.StopIntegrationTestService";
+
+        private InteractiveWindow _csharpInteractiveWindow;
+        private DTE _dte;
+        private EditorWindow _editorWindow;
+        private Process _hostProcess;
+        private IntegrationService _service;
+        private IpcClientChannel _serviceChannel;
+        private SolutionExplorer _solutionExplorer;
+        private string _serviceUri;
+        private bool _requireNewInstance;
+
+        static IntegrationHost()
+        {
+            // Enable TraceListenerLogging here since we can have multiple instances of IntegrationHost created per process, but this is a one-time setup
+            Trace.Listeners.Clear();
```

Comment:
But this is in the xunit process, not the target process, right? We already configure that for all our tests directly with xunit config. Having a library randomly clear this isn't good practice for a library.